src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -138,8 +138,11 @@ public object ReadValue()
                     _cancellationToken,
                     TaskCreationOptions.LongRunning,
                     TaskScheduler.Default);
-                task.Wait(_cancellationToken);
-                value = task.Result;
+
+                // We must not proceed until the additional task completes. After returning from a read, the underlying
+                // stream providing access to raw memory will be closed; if this occurs before the separate thread
+                // completes its read then an access violation can occur attempting to read from unmapped memory.
+                value = task.GetAwaiter().GetResult();
```

Comment:
Can we explicitly mention that waiting with cancellation would be a problem if the underlying task doesn't support the cancellation?